pkg/kubelet/pod_workers_test.go

```diff
@@ -640,13 +640,13 @@ func TestUpdatePod(t *testing.T) {
 				startedAt:          time.Unix(3, 0),
 				terminatingAt:      time.Unix(3, 0),
 				terminatedAt:       time.Unix(6, 0),
-				gracePeriod:        30,
+				gracePeriod:        1,
```

Comment:
The described behavior change in the test looks correct - if we observe a CREATE
while the pod is still terminating, we should be indicating to the CRI that the pod definition has likely changed (and we didn’t observe a force delete).  However, the test coverage here seems inadequate to convince me other scenarios might not be broken.

In particular, static pods may depend on the described behavior (because static pods can use a UID that doesn’t change on restart, but still need graceful termination).  It might be that we have to adapt this fix to differentiate between those two scenarios, and ensure static pod behavior is preserved.

Reply:
Thank you for your comments. I'll look into static pod behavior.